packages/terra-functional-testing/src/terra-cli/wdio/test-runner.js

```diff
@@ -29,39 +35,86 @@ class TestRunner {
    * @param {array} options.spec - A list of spec file paths.
-   * @param {Object} options.launcherOptions - Custom configuration launcher options. i.e. keepAliveSeleniumDockerService; updateScreenshots.
+   * @param {array} options.suite - Overrides specs and runs only the defined suites.
+   * @param {string} options.theme - A theme for the test run.
+   * @param {boolean} options.updateScreenshots - Updates all reference screenshots with the latest screenshots.
    * @returns {Promise} A promise that resolves with the test run exit code.
    */
   static async run(options) {
     let exitCode;
 
     try {
       const {
+        assetServerPort,
+        browsers,
         config,
+        disableSeleniumService,
+        externalHost,
+        externalPort,
         formFactor,
+        gridUrl,
+        keepAliveSeleniumDockerService,
         locale,
+        site,
         theme,
-        ...additionalLauncherOptions // hostname, port, baseUrl, suite, spec, launcherOptions
+        updateScreenshots,
+        ...cliOptions // spec, suite,
       } = options;
 
-      process.env.LOCALE = locale;
-      process.env.THEME = theme;
-
-      if (formFactor) {
-        process.env.FORM_FACTOR = formFactor;
-      }
+      const defaultWebpackPath = path.resolve(process.cwd(), 'webpack.config.js');
+
+      const launcherOptions = {
+        ...cliOptions,
+        baseUrl: `http://${externalHost || getIpAddress()}:${externalPort || 8080}`,
+        capabilities: getCapabilities(browsers, !!gridUrl),
+        hostname: gridUrl || 'localhost',
+        port: gridUrl ? 80 : 4444,
+        services: [
+          [TerraService, {
+            ...formFactor && { formFactor },
+            ...theme && { theme },
+          }],
+          [AssetServerService, {
+            ...locale && { locale },
+            ...assetServerPort && { port: assetServerPort },
+            ...site && { site },
+            ...theme && { theme },
+            ...fs.existsSync(defaultWebpackPath) && { webpackConfig: defaultWebpackPath },
+          }],
+          [VisualRegressionLauncher, {
+            ...locale && { locale },
+            ...theme && { theme },
+            ...updateScreenshots && { updateScreenshots },
+          }],
+          // Do not add the docker service if disabled.
+          ...(disableSeleniumService || gridUrl ? [] : [[SeleniumDockerService]]),
+        ],
+        launcherOptions: {
+          disableSeleniumService: disableSeleniumService || !!gridUrl,
+          ...assetServerPort && { assetServerPort },
+          ...formFactor && { formFactor },
+          ...keepAliveSeleniumDockerService && { keepAliveSeleniumDockerService },
+          ...locale && { locale },
+          ...site && { site },
+          ...theme && { theme },
+          ...updateScreenshots && { updateScreenshots },
```

Comment:
It doesn't hurt anything to expose all of these, but I think limiting it to only what we need might reduce some code here. I think now only the onComplete reads this and it only needs theme, locale, form factor